modules/layers/src/icon-layer/icon-layer.js

```diff
@@ -132,9 +132,10 @@ export default class IconLayer extends Layer {
     const {iconAtlas, iconMapping, data, getIcon} = props;
 
     let iconMappingChanged = false;
+    const prePacked = iconAtlas || this.props._asyncPropOriginalValues.iconAtlas;
```

Comment:
@ibgreen I really don't like this hack. Basically we need to determine here whether an async prop is pending or undefined.

Reply:
>  Basically we need to determine here whether an async prop is pending or undefined.

The problem is that we don't want to incur the cost of autopacking if an atlas is loading?

Ideas?
- The layer could offer an API to inspect status of async props `layer.isAsyncPropPending(propName)` 
- offer API to check whether prop values were supplied `layer.hasProp(propName)`
- Add a `pendingValue` to the async prop definition?